services/121-service/src/payments/payments.module.ts

```diff
@@ -71,4 +74,25 @@ import { createScopedRepositoryProvider } from '@121-service/src/utils/scope/cre
     PaymentsProgressHelperService,
   ],
 })
-export class PaymentsModule {}
+export class PaymentsModule implements OnModuleInit {
+  private readonly logger = new Logger(PaymentsModule.name);
+
+  onModuleInit() {
```

Comment:
so when does this run? on application startup, or is it lazy-loaded whenever the payment module is used?

Either way, it seems that these env-variables are mostly needed for 'do payment', while the payment module has wider use than that (e.g. getting payments/transactions). Shouldn't we tie it more to that by doing this at payment creation or something? Actually, I'm now reading the original task-title from Ruben where he suggested to do this in payment-execution service, which means at actually starting/retrying the payment. In that case we could (potentially) also handle errors by letting transactions fail, which is a very standard and clear pattern for us and our users. In your current case, what would a failure look like?

Reply:
Bit of copypasta from another reply:

- I wanted the application to **not** start when we have missing FSP variables, basically fail early and loud and at the moment when a developer is already focused on the instance/application. Failing when a payment is created and then having an end user going through all the channels to end up at a developer to fix this is just a lot of unnecessary overhead imho.
- using `onModuleInit` on the PaymentsModule seems the right place to hook into application startup

Let's discuss

Reply:
Agreed with Jannis to keep this.